src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.cs

```diff
@@ -0,0 +1,574 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Versioning;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.NuGetPackageDownloader;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Installer.Windows;
+using Microsoft.DotNet.ToolPackage;
+using Microsoft.DotNet.Workloads.Workload.Install.InstallRecord;
+using Microsoft.Extensions.EnvironmentAbstractions;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using Microsoft.Win32.Msi;
+using NuGet.Versioning;
+using static Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver;
+
+namespace Microsoft.DotNet.Workloads.Workload.Install
+{
+    [SupportedOSPlatform("windows")]
+    internal class NetSdkMsiInstallerClient : MsiInstallerBase, IWorkloadPackInstaller
```

Comment:
this class should be testable by mocking out the windows dependencies

Reply:
and we should have the brownbag on how we test this class with server class

Reply:
This class relies heavily on Windows specific operations, including IPC, the package cache and subsequent ACLs, registry and MSI APIs in the OS. I'll sync with you offline